### ISSUE TITLE
"ConfigException: Found a listener without a port" even though port is set

### ISSUE THREAD

**Issue by mandrean** [confluentinc/rest-utils#68]
## Issue
Getting `ConfigException: Found a listener without a port` even though a listener with a port is clearly set.

## Steps to reproduce
Set a listener with a port, for example `confluent-registry-0.confluent-registry-listeners.default.svc.cluster.local:8081`

## Error message
```
[2017-06-12 14:20:40,890] ERROR Server died unexpectedly:  (io.confluent.kafka.schemaregistry.rest.SchemaRegistryMain)
io.confluent.common.config.ConfigException: Found a listener without a port. All listeners must have a port. The listener without a port is: confluent-registry-0.confluent-registry-listeners.default.svc.cluster.local:8081
	at io.confluent.rest.Application.parseListeners(Application.java:350)
	at io.confluent.kafka.schemaregistry.storage.KafkaSchemaRegistry.getPortForIdentity(KafkaSchemaRegistry.java:196)
	at io.confluent.kafka.schemaregistry.storage.KafkaSchemaRegistry.<init>(KafkaSchemaRegistry.java:129)
	at io.confluent.kafka.schemaregistry.rest.SchemaRegistryRestApplication.setupResources(SchemaRegistryRestApplication.java:53)
	at io.confluent.kafka.schemaregistry.rest.SchemaRegistryRestApplication.setupResources(SchemaRegistryRestApplication.java:37)
	at io.confluent.rest.Application.createServer(Application.java:149)
	at io.confluent.kafka.schemaregistry.rest.SchemaRegistryMain.main(SchemaRegistryMain.java:43)
```

## Caused by
No sanity checking of wether a URI scheme has been provided before trying to get the URI port, see [Application.java#L382](https://github.com/confluentinc/rest-utils/blob/master/core/src/main/java/io/confluent/rest/Application.java#L382)

Quick and dirty REPL proof:
<img width="375" alt="skarmavbild 2017-06-12 kl 19 36 31" src="https://user-images.githubusercontent.com/4631744/27046912-83e3c1b6-4fa6-11e7-9b5f-a5bebdc59c23.png">

**Comment by mandrean** [comment]
PR: https://github.com/confluentinc/rest-utils/pull/69

**Comment by ewencp** [comment]
Fixed by #69, which was merged to 3.4.x and master.
